### PR TITLE
Test merges fixups

### PR DESCRIPTION
**Pull request by dmose**
Running a bunch of pushes to pine in preparation for setting this up in cron, and needing to debug various things, it became clear that various things weren't aborting on errors like they should have. There are several small bugs in play here; this PR fixes them, and renames some variables for clarity.

### CODE REVIEW

File: bin/download-firefox-artifact [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 # This looks for a mozilla-central artifact build as a sibling of the
 # activity-stream tree.  If it's not there, it creates it.  If it is there, it
 # updates it.
 
-ROOT=`dirname $0`
-FIREFOX_PATH="$ROOT/../../mozilla-central"
+# If AS_GIT_BIN_REPO (the git repo from which prepare-mochitests-dev and
+# friends will be executed) isn't set in the environment, just use the repo
+#  we're running from.
+if [ -z ${AS_GIT_BIN_REPO+x} ]; then
```
Comment (Mardak) [inline bin/download-firefox-artifact]: Eh, I suppose if we tried to refactor to share this with `prepare-mochitest-dev`, that shared script would also need its own set of detecting which version to use. I guess this is fine for now.

Comment (Mardak) [inline bin/download-firefox-artifact]: Also just making sure, these environment variables are only indirectly set when `prepare-mochitest-dev` is called where `test-merges` is passing it along?

File: bin/test-merges.js [hunk 2]
```diff
@@ -155,8 +155,8 @@ function exportToLocalMC(commitId) {
     // use echo.
     shelljs.exec(`
       echo yes | \
-      env AS_PMD_GIT_DIR=${AS_PMD_GIT_DIR} SYMLINK_TESTS=false ENABLE_MC_AS=1 \
-      bash ${PREPARE_MOCHITESTS_DEV}`,
+      env AS_GIT_BIN_REPO=${AS_GIT_BIN_REPO} SYMLINK_TESTS=false ENABLE_MC_AS=1 \
+      ${PREPARE_MOCHITESTS_DEV}`,
```
Comment (Mardak) [inline bin/test-merges.js]: nit: could you indent these to better indicate they're part of one big command even though the `\` and `exec` imply that?

File: bin/download-firefox-artifact (hunk 1 above)
Comment (dmose) [inline bin/download-firefox-artifact]: Usually, AS_GIT_BIN_REPO is initialized to TESTING_LOCAL_GIT in test-merges.js.  That said, if you want to debug from your own git repo instead, you'll be setting AS_GIT_BIN_REPO in the environment that test-merges.js is started under.   Whatever the value is gets used to find prepare-mochitest-dev, and is passed down via the environment to prepare-mochitest-dev

File: bin/test-merges.js (hunk 2 above)
Comment (dmose) [inline bin/test-merges.js]: Sure.

**Comment by as-pine-proxy** [comment]
[Treeherder: 027dbbc70081da98012e36237440df2207116ec4](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=027dbbc70081da98012e36237440df2207116ec4)

**Pull request merged**